### ISSUE TITLE
CPU MaxPool2d is very slow

### ISSUE THREAD

**Issue by dashesy** [pytorch/pytorch#28733]
## 🐛 Bug

MaxPool2d is too slow, it is better to replace it with Conv2d and stides

## To Reproduce

```python
import torch
import time
import numpy as np
import torch.nn as nn

model = nn.Conv2d(in_channels=16, out_channels=16, kernel_size=3, stride=2)
net = nn.MaxPool2d(kernel_size=3, stride=2)
blob = torch.randn(1, 16, 3000, 2000, device='cpu')

t0 = time.time()
with torch.no_grad():
    outputs = model(blob)
print("PyTorch Conv: {}".format(time.time() - t0))

t0 = time.time()
with torch.no_grad():
    pred = net(blob)
print("PyTorch MaxPool: {}".format(time.time() - t0))

# compare with ONNX
import onnx
import onnxruntime as rt

targets = ['result']
onnxfile = "/mnt/output/gr/conv.onnx"
torch.onnx.export(model, blob, onnxfile,
                  input_names=['data'],
                  output_names=targets)

onnxfile2 = "/mnt/output/gr/max.onnx"
torch.onnx.export(net, blob, onnxfile2,
                  input_names=['data'],
                  output_names=targets)

sess = rt.InferenceSession(onnxfile)
t0 = time.time()
outputs_ort = sess.run(targets, {
    'data': blob.numpy()
})
print("ONNX Conv: {}".format(time.time() - t0))

sess = rt.InferenceSession(onnxfile2)
t0 = time.time()
pred_ort = sess.run(targets, {
    'data': blob.numpy()
})
print("ONNX Conv: {}".format(time.time() - t0))
```

> PyTorch Conv: 0.1909499168395996
> PyTorch MaxPool: 0.8006699085235596
> ONNX Conv: 0.3233060836791992
> ONNX Conv: 0.26441049575805664

## Expected behavior

For MaxPool2d to have performance that is within the same range as convolution

## Environment

PyTorch version: 1.3.0
Is debug build: No
CUDA used to build PyTorch: 10.1.243

OS: Ubuntu 16.04.6 LTS
GCC version: (Ubuntu 5.4.0-6ubuntu1~16.04.11) 5.4.0 20160609
CMake version: version 3.14.0

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: GeForce GTX 1070
Nvidia driver version: 418.40.04
cuDNN version: /usr/local/lib/libcudnn.so.5.1.10

Versions of relevant libraries:
[pip] numpy==1.17.2
[pip] torch==1.3.0
[pip] torchvision==0.4.1a0+d94043a
[conda] Could not collect

**Comment by karimhasebou** [comment]
I will attempt to address this issue

**Comment by musikisomorphie** [comment]
Hi @VitalyFedyunin, is anyone still looking into this problem? If not, I can give it a try.

**Comment by karimhasebou** [comment]
> Hi @VitalyFedyunin, is anyone still looking into this problem? If not, I can give it a try.

Go ahead. I gave it a try but didn't have any luck.

**Comment by musikisomorphie** [comment]
> > Hi @VitalyFedyunin, is anyone still looking into this problem? If not, I can give it a try.
> 
> Go ahead. I gave it a try but didn't have any luck.

Okay, I will see what I can do.

**Comment by gchanan** [comment]
Addressed by https://github.com/pytorch/pytorch/issues/28733.

**Comment by kumar-akshay324** [comment]
Anything on this guys? Do we have a better alternative to using MaxPool2D? We are running into the same issue in the C++ version.

**Comment by jgong5** [comment]
> Anything on this guys? Do we have a better alternative to using MaxPool2D? We are running into the same issue in the C++ version.

This is an old issue. Have you tried the latest MaxPool2D? In particular, we recommend use channels last format since we can vectorize on the channel dim if data is contiguous on the channel.
